src/components/Form.js

```diff
@@ -0,0 +1,114 @@
+import { useRouter } from "next/router";
+
+export default function Form({
+                               onSubmitForm,
+                               initialData = {
+                                 transactionType: "",
+                                 amount: "",
+                                 currency: "EUR",
+                                 date: new Date().toISOString().substring(0, 10),
+                                 category: "",
+                                 paymentMethod: "",
+                                 description: "",
+                               },
+                               formTitle,
+                               confirmButtonText,
+                               add,
+                               edit,
+                             }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    data.amount = parseFloat(data.amount).toFixed(2);
+    onSubmitForm(data);
+  }
+
+  function handleAmountInput(event) {
+    const value = event.target.value;
+    const regex = /^(?!-)\d+(\.\d{0,2})?$/;  // Disallow certain characters & negative values
+    if (!regex.test(value)) {
+      event.target.value = value.slice(0, -1);
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <div>
+        {add && (<button type="button" onClick={() => {
```

Comment:
"add" and "edit" might be misleading / misunderstood. I would recommend to rename it e.g. "addMode" or "addTransaction"

Reply:
Changed accordingly